lib/nmatrix/io/csv_reader.rb

```diff
@@ -0,0 +1,153 @@
+#--
+# = NMatrix
+#
+# A linear algebra library for scientific computation in Ruby.
+# NMatrix is part of SciRuby.
+#
+# NMatrix was originally inspired by and derived from NArray, by
+# Masahiro Tanaka: http://narray.rubyforge.org
+#
+# == Copyright Information
+#
+# SciRuby is Copyright (c) 2010 - 2014, Ruby Science Foundation
+# NMatrix is Copyright (c) 2012 - 2014, John Woods and the Ruby Science Foundation
+#
+# Please see LICENSE.txt for additional copyright notices.
+#
+# == Contributing
+#
+# By contributing source code to SciRuby, you agree to be bound by
+# our Contributor Agreement:
+#
+# * https://github.com/SciRuby/sciruby/wiki/Contributor-Agreement
+#
+# == io/csv_reader.rb
+#
+# a csv file reader.
+#
+#++
+
+# TODO: ADD quote sepration
+module NMatrix::IO::Csv
+  # Reader for a version .csv file.
+
+  class << self
+    # call-seq:
+    #     load(filename) -> NMatrix
+    #     load(filename, options) -> NMatrix
+    #
+    # Load a Comma Seperated Value CSV file as a matrix.
+    #
+    # * *Arguments* :
+    #   - +filename+ -> String giving the name of the file to be loaded.
+    #   - +options+ -> Hash with options (:delim String, ',' by default; :dtype Symbol, 
+    #   may be the type supported by SciRuby, :float64 by default)
+    #
+    # * *Returns* :
+    #   - A NMatrix::IO::Csv::CsvReader object 
+    #   with the matrix stored in its +matrix+ property
+    #
+    # * *Using* *options* :
+    #   load('mat.csv', delim: ',')
+    #   load('mat.csv', delim: ',' , dtype: :int32)
+    #
+    #   PLEASE NOTE THAT: 
+    #   specifying :dtype with :object will result in a :float64 cast
+    #
+    def load(filename, options = {})
+      CsvReader.new(filename, options).matrix
+    end
+  end
+
+  class CsvReader #:nodoc:
+    @@csv_plugged = true
+    if File.exist?('lib/nmatrix/csv.rb')
```

Comment:
This will not work when the gem is published. Do you how the `.gemspec` file creates your gem?

It adds the files that are given to the `files` variable. The `lib/nmatrix/csv.rb` file is never added to nmatrix and this will always when installing the gems individually.